integration/pvtdata/implicit_coll_test.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pvtdata
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/hyperledger/fabric/integration/chaincode/kvexecutor"
+	"github.com/hyperledger/fabric/integration/nwo"
+	"github.com/hyperledger/fabric/integration/nwo/commands"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/tedsuo/ifrit"
+)
+
+var _ bool = Describe("PrivateData implicit collection", func() {
+	var (
+		network       *nwo.Network
+		process       ifrit.Process
+		orderer       *nwo.Orderer
+		testChaincode chaincode
+	)
+
+	BeforeEach(func() {
+		By("setting up the network")
+		network = initThreeOrgsSetup(false)
+
+		By("setting the pull retry threshold to 0 on all peers")
+		// set pull retry threshold to 0 to disable pulling
+		for _, p := range network.Peers {
+			core := network.ReadPeerConfig(p)
+			core.Peer.Gossip.PvtData.PullRetryThreshold = 0
+			network.WritePeerConfig(p, core)
+		}
+
+		By("starting the network")
+		process, orderer = startNetwork(network)
+
+		By("deploying new lifecycle chaincode")
+		testChaincode = chaincode{
+			Chaincode: nwo.Chaincode{
+				Name:        "kvexecutor",
+				Version:     "1.0",
+				Path:        components.Build("github.com/hyperledger/fabric/integration/chaincode/kvexecutor/cmd"),
+				Lang:        "binary",
+				PackageFile: filepath.Join(network.RootDir, "kvexcutor.tar.gz"),
+				Label:       "kvexcutor",
+				Sequence:    "1",
+			},
+			isLegacy: false,
+		}
+		nwo.EnableCapabilities(network, channelID, "Application", "V2_0", orderer, network.Peers...)
+		deployChaincode(network, orderer, testChaincode)
+	})
+
+	AfterEach(func() {
+		testCleanup(network, process)
+	})
+
+	It("disseminates implicit collection data owned by the peer's org ", func() {
+		peer1 := network.Peer("Org1", "peer0")
+		peer2 := network.Peer("Org2", "peer0")
+
+		By("writing private data to org1's and org2's implicit collections")
+		writeInput := []kvexecutor.KVData{
+			{Collection: "_implicit_org_Org1MSP", Key: "org1_key1", Value: "org1_value1"},
+			{Collection: "_implicit_org_Org2MSP", Key: "org2_key1", Value: "org2_value1"},
+		}
+		writeImplicitCollection(network, orderer, testChaincode.Name, writeInput, peer1, peer2)
```

Comment:
Since Org1 and Org2 implicit collection data is written to both Org1 and Org2 peers, we'd expect the private data to be disseminated to the authorized peers, both before and after this change.
Maybe I'm missing something, but does this test actually test that the dissemination is suppressed on peers that are not part of the implicit collection?

I was expecting to see maybe endorsement on only one of the two orgs, then we could verify that the other org didn't receive the private data.

Reply:
Talked to Dave offline. https://jira.hyperledger.org/browse/FAB-17463 is needed to finish the verification (when we can configure different requiredPeerCount/maxPeerCount values for different peers). Therefore, the test will be enhanced/finalized in FAB-17463.

Reply:
ok, i've verified for myself that the new logic is good, will look for the final integration test in the next PR.